pandas/core/arrays/arrow/dtype.py

```diff
@@ -132,6 +132,14 @@ def type(self):
             return time
         elif pa.types.is_decimal(pa_type):
             return Decimal
+        elif pa.types.is_dictionary(pa_type):
+            from pandas.core.arrays import Categorical
+
+            return Categorical
```

Comment:
id be OK either way.  `CategoricalDtypeType` seems like the safe option internal-consistency wise, but seems pretty useless for a user

Reply:
Alternatively, would it be okay to return a Categorical here? The "scalar" could be a length 1 `Categorical` in practice but I'm not sure if that breaks the mental model of anything else

Reply:
That seems really weird to me.  Is there some test(s) that are easier to fix with that than with either of the other options?

We should probably deprecate/change CategoricalDtype.type at some point and ensure these match.

Reply:
Not sure if this makes other tests easier as of yet since we don't test `pa.dictionary` fully yet.

Will just use ` CategoricalDtypeType` here for now.